### PR TITLE
ENH: Accuracy feature for all feeds

### PR DESCRIPTION
**Pull request by mauroasilva**
This pull request adds an accuracy feature that is meant to represent the amount of false positives the feed has. Say you have a feed with 10 false positives for every 100 events, than you would give it a 90 in the accuracy parameter (as in 90%).

The reason I put a quantitative value (0 to 100), instead of a qualitative value (low, medium, high), was because I was thinking that, in the future, a feedback loop could be done in which the operators give feedback to the system regarding whether a given event is correctly identified or not, and that could be used to automatically calculate the feed's accuracy.

Although it is not hard to change it into a qualitative value if you guys prefer. What do you think is the best option?

### CODE REVIEW

File: intelmq/conf/harmonization.conf [hunk 1]
```diff
@@ -311,6 +311,10 @@
             "description": "The URL of a given abuse feed, where applicable",
             "type": "URL"
         },
+        "feed.accuracy": {
```
Comment (sebix) [inline intelmq/conf/harmonization.conf]: You also need it for events, not just reports.

Comment (SYNchroACK) [inline intelmq/conf/harmonization.conf]: @sebix in your opinion we should use the same procedure like the following?
https://github.com/certtools/intelmq/blob/master/intelmq/lib/message.py#L221

Comment (mauroasilva) [inline intelmq/conf/harmonization.conf]: @sebix you are right... If you check the message.py it already has that field being passed on to the events, but I forgot to put it into the harmonization document.

Comment (SYNchroACK) [inline intelmq/conf/harmonization.conf]: @sebix should we change the unittests to be compliant?

**Comment by sebix** [comment]
- tests for the Accuracy type are missing
- Is the accuracy value mandatory for all collectors? Your modifications in the collectors depend on the existance of `self.parameters.accuracy`. If it does not exist, the bot will fail.

        AttributeError: 'Parameters' object has no attribute 'accuracy'
  - If yes, the accuracy can't be defined when spawning new bots with the manager, as accuracy is not defined in `intelmq/bots/BOTS`
  - If no, should there be a default value if nothing is given? Or does the value not exist in the reports and events?
- Please generate a new `docs/Harmonization-fields.md` using `intelmq/bin/intelmq_gen_harm_docs.py`
- document the addition in the changelog

File: intelmq/conf/harmonization.conf (hunk 1 above)
Comment (sebix) [inline intelmq/conf/harmonization.conf]: The unittests are missing a check for the Event(report) feature, yes.

**Comment by sebix** [comment]
Ok, fine for me. As it's assigned to you, feel free to merge now.

Please also note that you can always easily correct the last commit by using the `--amend` flag: `git commit --amend`.

**Comment by mauroasilva** [comment]
Ok thanks for the heads up.

**Comment by aaronkaplan** [comment]
Hi folks, especially @mauroasilva : did you see this ? https://github.com/MISP/misp-taxonomies . Especially the Admiralty Scale seems reasonable for me for "accuracy".
https://github.com/MISP/misp-taxonomies/tree/master/admiralty-scale

What do you think? Should we quantify the information into A,B,C,D,E,F? It would make sense to build ontop of existing schemas.

**Comment by robcza** [comment]
Other scale being used is TLP https://www.us-cert.gov/tlp
For compatibility reasons, there should be straightforward guide to convert form 0-100 to TLP or Admiralty Scale, yet IntelMQ could use 0-100 in the background as it easier to handle with different score factors influencing the final value.

**Comment by mauroasilva** [comment]
@aaronkaplan that looks nice indeed and it is good that it is already a "standard". I begun this PR exactly by asking opinions about wether we should use qualitative values (like the one you described) or quantitative. The reason why I chose quantitative as I explained previously was to be able to have a kind of false positive percentage let's call it. Which could be calculated automatically based on feedback gathered by operators. I can change it if you guys feel it is the best approach, but that was my mindset when I implemented this feature.

@robcza I think TLP will eventually have to be added to IntelMQ but it is not really an accuracy measure, but a disclosure "range".

**Comment by wagner-certat** [comment]
@robcza How is the traffic light protocol related to accuracy?

**Comment by robcza** [comment]
@wagner-certat that was a misunderstanding as @mauroasilva pointed out

**Pull request merged**